src/Lykke.Job.HistoryExportBuilder.AzureRepositories/ExpiriesRepository/ExpiryEntryRepository.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using AzureStorage;
+using AzureStorage.Tables.Templates.Index;
+using Lykke.Job.HistoryExportBuilder.Core.Domain;
+using Lykke.Job.HistoryExportBuilder.Core.Services;
+
+namespace Lykke.Job.HistoryExportBuilder.AzureRepositories.ExpiriesRepository
+{
+    public class ExpiryEntryRepository : IExpiryWatcher
+    {
+        public const string TableName = "HistoryExportExpiryEntries";
+
+        private readonly INoSQLTableStorage<ExpiryEntryEntity> _tableStorage;
+
+        public ExpiryEntryRepository(
+            INoSQLTableStorage<ExpiryEntryEntity> tableStorage)
+        {
+            _tableStorage = tableStorage;
+        }
+
+        public async Task AddAsync(IExpiryEntry entry)
+        {
+            var alreadyExisted = !await
+                _tableStorage.TryInsertAsync(
+                    new ExpiryEntryEntity
+                    {
+                        PartitionKey = entry.ClientId,
```

Comment:
Adde separate static method to ExpiryEntryEntity - to create partition key from ClientId